platform/tm4c/export/platform/execution.hpp

```diff
@@ -178,7 +184,82 @@ static inline void wfe()
     asm volatile ("wfe");
 }
 
-} // namespace ecl
+#if USE_SYSTMR
+
+namespace systmr
+{
+
+//! Represents the minimum and the maximun possible values for SYSTMR frequency.
+//! This allows to guarantee the correct SYSTRM frequency regardless of SystemCoreClock.
+static const uint32_t systmr_freq_min = 20;
+static const uint32_t systmr_freq_max = 1000;
+
+static_assert((THECORE_CONFIG_SYSTMR_FREQ >= systmr_freq_min) &&
+               (THECORE_CONFIG_SYSTMR_FREQ <= systmr_freq_max),
+               "The value of the THECORE_CONFIG_SYSTMR_FREQ frequency is out of the range.");
+
+ //! Enables timer, start counting
+static inline void enable()
+{
+    uint32_t reload_val = (1000 / THECORE_CONFIG_SYSTMR_FREQ) * clk_spd() / 1000;
+
+    SysTickPeriodSet(reload_val);
+
+    // reset SysTick counter
+    HWREG(NVIC_ST_CURRENT) = 0;
+
+    SysTickIntEnable();
+    SysTickEnable();
+}
+
+//! Disables timer, stop counting
+static inline void disable()
+{
+    SysTickDisable();
+    SysTickIntDisable();
+}
+
+//! Gets speed in which timer generate events.
+//! \details _Each event generates interrupt_, but not necessarily timer interrupt
+//! is passed to the user. In other words, each event will wake-up processor.
+//! Useful in conjunction with WFI.
+//! \retval System timer events generation speed in Hz.
+static inline auto speed()
+{
+    uint32_t reload_val = (1000 / THECORE_CONFIG_SYSTMR_FREQ) * clk_spd() / 1000;
+
+    return clk_spd() / reload_val;
```

Comment:
Let's drop `clk_*` usage here, remove `clk_*` routines from https://github.com/forGGe/theCore/pull/273/files#diff-d49fc3d45839650098c3315855c8a367R148 and below and also drop `clk_*` from execution module:
https://github.com/vadimol/theCore/blob/issue-247-part2/platform/common/export/common/execution.hpp#L73 

Since every platform at this moment provides `spin_wait` API, no need to use vague DWT.
`#define THECORE_PLATFORM_STUB_DEFINE_SPIN_WAIT` and DWT initialization code should be removed as well.